pkgs/by-name/co/cosmic-player/package.nix

```
@@ -0,0 +1,95 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  rustPlatform,
+  autoPatchelfHook,
+  makeBinaryWrapper,
+  alsa-lib,
+  cosmic-icons,
+  ffmpeg,
+  llvmPackages,
+  just,
+  pkg-config,
+  libxkbcommon,
+  wayland,
+  xorg,
+}:
+
+rustPlatform.buildRustPackage {
+  pname = "cosmic-player";
+  version = "0-unstable-2024-06-10";
+
+  src = fetchFromGitHub {
+    owner = "pop-os";
+    repo = "cosmic-player";
+    rev = "afcb8c72bc0fc4022ebc68393056e9578bfe8db0";
+    hash = "sha256-JEKEZ3Vb41CKKCwd/AJam26GgfUIGH3kc5kzOiinsj8=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+    outputHashes = {
+      "accesskit-0.12.2" = "sha256-ksaYMGT/oug7isQY8/1WD97XDUsX2ShBdabUzxWffYw=";
+      "atomicwrites-0.4.2" = "sha256-QZSuGPrJXh+svMeFWqAXoqZQxLq/WfIiamqvjJNVhxA=";
```

```suggestion
  useFetchCargoVendor = true;
  cargoHash = "";
```
Please check whether you can use the new fetcher. Then you do not need to copy Cargo.lock here.